cmd/export_ledger_entry_changes.go

```diff
@@ -52,49 +66,78 @@ be exported.`,
 		}
 
 		accChannel, offChannel, trustChannel := createChangeChannels(exportAccounts, exportOffers, exportTrustlines)
-		go input.StreamChanges(core, startNum, endNum, accChannel, offChannel, trustChannel)
-
-		transformedAccounts, transformedOffers, transformedTrustlines := input.ReceiveChanges(accChannel, offChannel, trustChannel, cmdLogger)
-
-		// TODO: add export functionality that periodically exports transformed data in batch_size increments instead of printing at the end
-		fmt.Println(transformedAccounts)
-		fmt.Println(transformedOffers)
-		fmt.Println(transformedTrustlines)
-
-		/*
-			1. Instantiate a captive core instance - DONE
-				a) If the start and end are provided, then use a bounded range and exit after exporting the info inside the range
-				b) If the end is omitted, use an unbounded range and continue exporting as new ledgers are added to the network
-			2. Call GetLedger() constantly in a separate goroutine - DONE
-				a) Create channels for each data type
-				b) Process changes for the ledger and send changes to the channel matching their type
-			3. On the other end, receive changes from the channel
-				a) Call transform on individual changes - DONE
-				b) Once batch_size ledgers have been sent, encode and export the changes - TODO
-		*/
+
+		go input.StreamChanges(core, startNum, endNum, batchSize, accChannel, offChannel, trustChannel, cmdLogger)
+		if endNum != 0 {
+			batchCount := int(math.Ceil(float64(endNum-startNum+1) / float64(batchSize)))
+			for i := 0; i < batchCount; i++ {
+				transformedAccounts, transformedOffers, transformedTrustlines := input.ReceiveChanges(accChannel, offChannel, trustChannel, cmdLogger)
+				exportTransformedData(outFile, useStdout, transformedAccounts, transformedOffers, transformedTrustlines)
+			}
+
+		} else {
+			batchNum := uint32(1)
+			for {
```

Comment:
I think this is an infinite loop? When would this terminate?

Reply:
This is the unbounded case where exports should continue happening indefinitely as the network updates.